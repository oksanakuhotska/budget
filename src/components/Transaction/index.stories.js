import React from 'react';

import Transaction from '.' ;

export default {
  title: 'Example/Transaction',
  component: Transaction,
};

const Template = (args) => <Transaction {...args} />;

export const Income = Template.bind({});
Income.args = {
  transaction: {
    comment: 'some income',
		value: 23,
		date: '22.02.2023',
  },
};
