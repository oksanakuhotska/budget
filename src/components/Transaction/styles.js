import styled from 'styled-components';

export const Wrapper = styled.div`
	background-color: ${({ value }) => value < 0 ? '#f8e9e9' : '#d6efdf'};
	border: 2px solid ${({ value }) => value < 0 ? '#f9f3db' : '#cbe5e8'};
	border-radius: 10px;
	padding: 10px;
	margin-bottom: 10px;
	display: flex;
	align-items: center;
`;
Wrapper.displayName = 'TransactionWrapper';

export const TransactionDate = styled.div`
	flex-grow: 1;
`;
TransactionDate.displayName = 'TransactionDate';

export const Value = styled.div`
	flex-grow: 1;
`;
Value.displayName = 'Value';

export const Comment = styled.div`
	flex-grow: 2;
`;
Comment.displayName = 'Comment';