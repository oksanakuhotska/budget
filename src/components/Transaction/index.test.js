import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Transaction from '.';

describe('Transaction component', () => {
  let props;

  beforeEach(() => {
    props = {
      transaction: {
        value: 23,
        comment: 'test',
        date: '22.02.2023'
      }
    };
  });

  it('should show transaction', () => {
    const { container } = render(<Transaction {...props} />);
    expect(container).toMatchSnapshot();
  });
});