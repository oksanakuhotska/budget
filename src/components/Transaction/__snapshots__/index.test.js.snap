// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component should show transaction 1`] = `
<div>
  <div
    class="sc-bcXHqe XuCUo"
    value="23"
  >
    <div
      class="sc-gswNZR ScxuZ"
    >
      22.02.2023
    </div>
    <div
      class="sc-dkrFOg gZAkGf"
    >
      23.00
    </div>
    <div
      class="sc-hLBbgP hRFeTc"
    >
      test
    </div>
  </div>
</div>
`;
